cmd/tikv-ctl/src/main.rs

```diff
@@ -796,12 +796,18 @@ fn compact_whole_cluster(
     threads: u32,
     bottommost: BottommostLevelCompaction,
 ) {
-    let stores = pd_client
+    let all_stores = pd_client
         .get_all_stores(true) // Exclude tombstone stores.
         .unwrap_or_else(|e| perror_and_exit("Get all cluster stores from PD failed", e));
 
+    let tikv_stores = all_stores.iter().filter(|s| {
+        !s.get_labels()
+            .iter()
+            .any(|l| l.get_key() == "engine" && l.get_value() == "tiflash")
```

Comment:
maybe you need to consider this code to check the store belongs the tiflash.
https://github.com/tikv/tikv/blob/312e0fb7f9f77e6002d0a336a58e84f3c4c12216/components/raftstore/src/store/worker/region.rs#L839-L842

Reply:
Sure - I think it's equivalent, but can use the constants. Is there a place that it would make sense to pull this out into a common utility function, or should I just copy and paste the code?

Reply:
yes, it's little different that the key and values need to lower case.